packages/firebase_storage/firebase_storage/lib/src/firebase_storage.dart

```diff
@@ -60,38 +55,40 @@ class FirebaseStorage extends FirebasePluginPlatform {
   ///
   /// If [app] is not provided, the default Firebase app will be used.
   /// If [bucket] is not provided, the default storage bucket will be used.
-  static FirebaseStorage /*!*/ instanceFor({
-    FirebaseApp app,
-    String bucket,
+  static FirebaseStorage instanceFor({
+    FirebaseApp? app,
+    String? bucket,
   }) {
     app ??= Firebase.app();
-    assert(app != null);
-
-    bucket ??= app.options.storageBucket;
 
-    // A bucket must exist at this point
-    // TODO(ehesp): Check whether `app.options.storageBucket` can be nullable post migration
-    if (bucket == null) {
+    if (bucket == null && app.options.storageBucket == null) {
```

Comment:
I updated this code to be a bit more clear about what is happening. `bucket` and `app.options.storageBucket` are nullable. If both are `null`, throw an error.

`_bucket` initialization occurs [here](https://github.com/FirebaseExtended/flutterfire/pull/4753/files#diff-eb1d5f7023d5b3aafdee60e56c19b2f594c1cebc44b9cfeed149fce706322750R74-R80)